spec/controllers/infra_networking_controller_spec.rb

```diff
@@ -1,27 +1,140 @@
+                                  :name   => "tag1",
+                                  :parent => classification)
+        tag2 = FactoryGirl.create(:classification_tag,
+                                  :name   => "tag2",
+                                  :parent => classification)
+        allow(Classification).to receive(:find_assigned_entries).with(switch).and_return([tag1, tag2])
+
+        tree_node_id = "sw-#{switch.id}"
+        seed_session_trees('infra_networking', :infra_networking_tree)
+        post :tree_select, :params => { :id => tree_node_id, :tree => :infra_networking_tree, :format => :js }
+
+        expect(response.status).to eq(200)
+        expect(response.body).to include('Department')
+        expect(response.body).to include(tag1.description)
+        expect(response.body).to include(tag2.description)
       end
 
-      it 'renders the network switch center toolbar' do
-        nodeid = [ems, cluster, switch].map { |item| TreeNode.new(item).key }.join('_')
-        expect(ApplicationHelper::Toolbar::InfraNetworkingCenter).to receive(:definition).and_call_original.at_least(:once)
-        post :tree_select, :params => { :id => nodeid }
+      describe 'switch' do
+        let(:ems) { FactoryGirl.create(:ems_vmware) }
+        let(:cluster) { FactoryGirl.create(:ems_cluster, :ems_id => ems.id) }
+
+        it 'renders the network switch center toolbar' do
+          nodeid = [ems, cluster, switch].map { |item| TreeNode.new(item).key }.join('_')
+          expect(ApplicationHelper::Toolbar::InfraNetworkingCenter).to receive(:definition).and_call_original.at_least(:once)
+          post :tree_select, :params => { :id => nodeid }
+        end
       end
     end
   end
+
+  context "#tags_edit" do
+    let!(:user) { stub_user(:features => :all) }
+    before(:each) do
+      EvmSpecHelper.create_guid_miq_server_zone
+      @ds = FactoryGirl.create(:switch, :name => "testSwitch")
+      allow(@ds).to receive(:tagged_with).with(:cat => user.userid).and_return("my tags")
+      classification = FactoryGirl.create(:classification, :name => "department", :description => "Department")
+      @tag1 = FactoryGirl.create(:classification_tag,
+                                 :name   => "tag1",
+                                 :parent => classification)
+      @tag2 = FactoryGirl.create(:classification_tag,
+                                 :name   => "tag2",
+                                 :parent => classification)
+      allow(Classification).to receive(:find_assigned_entries).with(@ds).and_return([@tag1, @tag2])
+      session[:tag_db] = "Switch"
+      edit = {
+        :key        => "Switch_edit_tags__#{@ds.id}",
+        :tagging    => "Switch",
+        :object_ids => [@ds.id],
+        :current    => {:assignments => []},
+        :new        => {:assignments => [@tag1.id, @tag2.id]}
+      }
+      session[:edit] = edit
+    end
+
+    after(:each) do
+      expect(response.status).to eq(200)
```

Comment:
Don't test implicitly in an `after` hook, it is intended to use only for database cleanup that you don't need due to `FactoryGirl`.